apps/frontend/src/features/login/EnterUsername.tsx

```diff
@@ -8,39 +8,45 @@ import { AppCopy } from "@/components/ui/AppCopy";
 import { SupportToast } from "@/components/ui/SupportToast";
 import { ERROR_SUPPORT_CONTACT } from "@/constants";
 
-interface EnterEmailProps {
-  submitEmail: (email: string) => Promise<void>;
+interface EnterUsernameProps {
+  submitUsername: (username: string) => Promise<void>;
 }
 
-const EnterEmail: React.FC<EnterEmailProps> = ({ submitEmail }) => {
-  const [email, setEmail] = useState("");
+const EnterUsername: React.FC<EnterUsernameProps> = ({ submitUsername }) => {
+  const [username, setUsername] = useState("");
   const [loading, setLoading] = useState<boolean>(false);
 
   const handleSubmit = async (e: React.FormEvent) => {
     e.preventDefault();
+    try {
+      UsernameSchema.parse(username);
+    } catch (error) {
+      console.error(error);
+      toast.error("Please enter a valid username");
```

Comment:
I liked the other error message slightly better because it's more explanatory (`"Username must be alphanumeric and between 3-20 characters"`).

Reply:
Or is this more vague because it can also take emails?